### PR TITLE
Add Flow

### LINKED ISSUE #2: Add flow

### PR DESCRIPTION
**Pull request by ferrannp**
Closes #2.

- Uses Flow that comes with latest `react-native init`

### CODE REVIEW

File: .flowconfig [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+suppress_type=$FlowFixMeProps
+suppress_type=$FlowFixMeState
+
+suppress_comment=\\(.\\|\n\\)*\\$FlowFixMe\\($\\|[^(]\\|(\\(<VERSION>\\)? *\\(site=[a-z,_]*react_native[a-z,_]*\\)?)\\)
+suppress_comment=\\(.\\|\n\\)*\\$FlowIssue\\((\\(<VERSION>\\)? *\\(site=[a-z,_]*react_native[a-z,_]*\\)?)\\)?:? #[0-9]+
+suppress_comment=\\(.\\|\n\\)*\\$FlowFixedInNextDeploy
+suppress_comment=\\(.\\|\n\\)*\\$FlowExpectedError
+
+[version]
+^0.78.0
```
Comment (tbergq) [inline .flowconfig]: can you add latest version of flow?

File: .flowconfig [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+[libs]
+node_modules/react-native/Libraries/react-native/react-native-interface.js
+node_modules/react-native/flow/
+node_modules/react-native/flow-github/
+
+[options]
+emoji=true
+
+esproposal.optional_chaining=enable
+esproposal.nullish_coalescing=enable
```
Comment (tbergq) [inline .flowconfig]: could you also add 

```
[lints]
unnecessary-optional-chain=error
```

File: .flowconfig (hunk 1 above)
Comment (ferrannp) [reply .flowconfig]: Not really, it gives several errors with RN. If you check my comment, I added the one that comes with `react-native init`. We should keep with that and update it when they update it IMO.

Comment (tbergq) [reply .flowconfig]: I prefer to be up to date and rather add the filepaths that gives errors to [untyped] but, I will allow it 🤔

**Pull request merged**